source/retryable-reads/tests/README.rst

```diff
@@ -0,0 +1,158 @@
+=====================
+Retryable Reads Tests
+=====================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+The YAML and JSON files in this directory tree are platform-independent tests
+that drivers can use to prove their conformance to the Retryable Reads spec.
+
+Prose tests, which are not easily expressed in YAML, are also presented
+in this file. Those tests will need to be manually implemented by each driver.
+
+Tests will require a MongoClient created with options defined in the tests.
+Integration tests will require a running MongoDB cluster with server versions
+4.2.0 or later. N.B. Strictly speaking, the server should implement the
+"Early Failure on Socket Disconnect" slated for 4.2.0, but this is not
+necessary for initial testing.
+
+
+Server Fail Point
+=================
+
+See: `Server Fail Point <../../transactions/tests#server-fail-point>`_ in the
+Transactions spec test suite.
+
+Disabling Fail Point after Test Execution
+-----------------------------------------
+
+After each test that configures a fail point, drivers should disable the
+``failCommand`` fail point to avoid spurious failures in
+subsequent tests. The fail point may be disabled like so::
+
+    db.runCommand({
+        configureFailPoint: "failCommand",
+        mode: "off"
+    });
+
+Network Error Tests
+===================
+
+Network error tests are expressed in YAML and should be run against a standalone,
+shard cluster, or single-node replica set. Until `SERVER-34943`_ is resolved,
+these tests cannot be run against a multi-node replicaset.
+
+.. _SERVER-34943: https://jira.mongodb.org/browse/SERVER-34943
+
+Test Format
+-----------
+
+Each YAML file has the following keys:
+
+- ``database_name`` and ``collection_name``: Optional. The database and
+  collection to use for testing.
+  
+- ``bucket_name``: Optional. The GridFS bucket name to use for testing.
```

Comment:
A few questions about the gridfs tests:
1. Can we provide more guidance on how to run the GridFS tests? We should mention how to create the GridFSBucket and that "data" will (optionally?) have "fs.files" and "fs.chunks".
1. What's the goal of "bucket_name"? Is it just to inform the runner that this is a gridfs test?
1. On that point, can you mention what should be the canonical way to tell that this is a gridfs test? Do we look for "gridfs" in the filename, the presence of "bucket_name", the presence of "fs.files" in "data"?